pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -997,10 +993,9 @@ protected File downloadSegmentFromDeepStore(SegmentZKMetadata zkMetadata)
         FileUtils.deleteQuietly(tempRootDir);
       }
     } finally {
+      _serverMetrics.addValueToTableGauge(_tableNameWithType, ServerGauge.SEGMENT_TABLE_DOWNLOAD_COUNT, -1);
```

Comment:
nit: just wondering if it might make sense to decrement after releasing like before? should be fine to keep this as well though